PRACK refactoring

maximilianfridrich opened the pull request:
- Add `awaiting_prack bool` to sipsess to be able to check whether the sipsess is still awaiting a PRACK to a 1xx response with SDP.
- Refactor PRACK logic to remove ht_prack which was unused and unneeded.
- Add logic to be conformant to RFC 3262.

sreimers commented on src/sipsess/update.c:
```diff
@@ -149,20 +151,25 @@ int sipsess_update(struct sipsess *sess)
+	if (!sess || sess->terminated || !sess->ctype || !sess->desc
+	   || (!sess->established && !sess->refresh_allowed)
+	   || sess->awaiting_answer)
 		return EINVAL;
```
Maybe we can split up these checks in EINVAL and EPROTO error parts? Makes debugging sometimes easier.

maximilianfridrich replied:
Makes sense, I split it up.

maximilianfridrich commented:
I am quite confused as to why some checks are failing. I rebased and added an additional test to retest. Maybe an issue with the build agent? Could someone trigger a re-check?

alfredh commented on src/sipsess/prack.c:
```diff
@@ -19,78 +19,147 @@
+	int err;
+
+	if (!req || req->tmr.th)
+		return -1;
```
better to return EINVAL here ...

maximilianfridrich replied:
Done.

alfredh commented on src/sipsess/sess.c:
```diff
@@ -321,3 +321,34 @@ void sipsess_abort(struct sipsess *sess)
+ * @return true if session is waiting for a PRACK to a 1xx containing SDP,
+ * 	   false otherwise
+ */
+bool sipsess_awaiting_prack(struct sipsess *sess)
```
can `sess` be const ?

maximilianfridrich replied:
Thank you. Done.

alfredh commented on src/sipsess/sess.c:
```diff
@@ -321,3 +321,34 @@ void sipsess_abort(struct sipsess *sess)
+ *
+ * @return True if a target refresh is currently allowed, otherwise false
+ */
+bool sipsess_refresh_allowed(struct sipsess *sess)
```
can `sess` be const ?

alfredh commented:
```
/home/runner/work/re/re/current/src/sipsess/sess.c:334:6: error: conflicting types for ‘sipsess_awaiting_prack’; have ‘_Bool(const struct sipsess *)’

```

alfredh commented:
looks good now. should we merge it to main?

maximilianfridrich commented:
I think @cspiel1 will look at it again later today, so maybe we can give him a little bit of time. The more reviewers the better.
